src/injectors-cache.ts

```diff
@@ -0,0 +1,156 @@
+/**
+ * Contains functions that inject dynamic content.
+ */
+
+import * as brotli from './brotli-wasm-wrapper';
+import {
+  CacheControl,
+  ContentEncoding,
+  HeaderKeys,
+  IncomingCloudflarePropertiesExtended,
+  supportsBrotli,
+} from './headers';
+
+const INJECTORS_CACHE_NAME = 'injectors-cache-v1';
+const textEncoder = new TextEncoder();
+
+/**
+ * Generates a complete cache key for a cache object.
+ *
+ * @param parts - unique list of identifiers for object in cache.
+ * @returns a unique cache key.
+ */
+function cacheKey(...parts: string[]): string {
+  return parts.join(';');
+}
+
+/**
+ * Generates a complete cache key for a Brotli-compressed cache object.
+ *
+ * @param parts - unique list of identifiers for object in cache.
+ * @returns a unique cache key.
+ */
+function cacheKeyBrotli(...parts: string[]): string {
+  return cacheKey(...parts, ContentEncoding.BROTLI);
+}
+
+/** */
+export async function hashObject(obj: unknown): Promise<string> {
+  const hashBuffer = await crypto.subtle.digest(
+    'SHA-1',
+    textEncoder.encode(JSON.stringify(obj))
+  );
+  return Array.from(new Uint8Array(hashBuffer))
+    .map((b) => b.toString(16).padStart(2, '0'))
+    .join('');
```

Comment:
Ah, I didn't realize we needed a hex string output. Given the difficulty here, we can go back to `hash.js` if you want.